parlai/core/torch_classifier_agent.py

```diff
@@ -168,6 +170,129 @@ def value(self) -> float:
             return numer / denom
 
 
+class AUCMetrics(Metric):
+    """
+    Computes Area Under ROC Curve (AUC) metrics.
+
+    Does so by keeping track of positives' and negatives' probability score counts in
+    Counters or dictionaries. Note the introduction of `max_bucket_dec_places`; this
+    integer number determines the number of digits to save for the probability scores. A
+    higher `max_bucket_dec_places` will a more accurate estimate of the exact AUC
+    metric, but may also use more space.
+    """
+
+    @property
+    def macro_average(self) -> bool:
+        """
+        Indicates whether this metric should be macro-averaged when globally reported.
+        """
+        return False
+
+    @classmethod
+    def raw_data_to_auc(
+        cls,
+        true_labels: List[int],
+        pos_probs: List[float],
+        class_name,
+        max_bucket_dec_places: float = 5,
```

Comment:
nit why is the default `max_bucket_dec_places` different from L210

Reply:
also nit: change the type of `max_bucket_dec_places` to `int` might be better?

Reply:
ahhh.... forgot to change; thanks!